lms/services/course.py

```diff
@@ -156,6 +165,34 @@ def get_courses(
             # Only select the ID of the deduplicated courses
         ).with_entities(Course.id)
 
+        if instructor_h_userid:
```

Comment:
Yes, this could be changed to a filter that takes an arbitrary tuple of (h_userid, role_type, role_scope) to make it more flexible, at least at this level.

He have to sort teacher and admins out first so let's revisit this when we have a better understand on what data we'd show for students.